src/views/CourtOrder.vue

```diff
@@ -0,0 +1,696 @@
+  }
+
+  get notationLabel (): string {
+    return this.isCourtOrder ? `${this.displayName} Text` : 'Court Order'
+  }
+
+  /** The notation textarea validation rules. */
+  get notationRules (): Array<(v) => boolean | string> {
+    return [
+      // Administrative Dissolution, Put Back On, Freeze/Unfreeze Business require a detailed comment
+
+      // Court Order requires a file or a comment
+      (v: string) => (
+        !!v ||
+        !this.isCourtOrder ||
+        (!!this.file && !!this.fileKey) ||
+        `Enter a ${this.displayName} and/or upload file`
+      ),
+      // others require a comment
+      (v: string) => (
+        !!v ||
+        (this.isCourtOrder) ||
+        `Enter a ${this.displayName}`
+      ),
+      (v: string) => (
+        !v ||
+        (v.length <= this.NOTATION_MAX_LENGTH) ||
+        'Maximum characters exceeded.'
+      )
+    ]
+  }
+
+  get isPageValid (): boolean {
+    const filingDataValid = (this.filingData.length > 0)
+    return (filingDataValid && this.courtOrderValid)
+  }
+
+  /**
+   * Routes to dashboard if there are no outstanding changes,
+   * else prompts user before routing.
+   */
+  goToDashboard (force = false): void {
+    // check if there are no data changes
+    if (!this.haveChanges || force) {
+      // route to dashboard
+      this.navigateToBusinessDashboard(this.getIdentifier)
+      return
+    }
+
+    // open confirmation dialog and wait for response
+    this.$refs.confirm.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Continue Out. Do you want to exit your filing?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Return to my filing',
+        no: null,
+        cancel: 'Exit without saving'
+      }
+    ).then(() => {
+      // if we get here, Yes was clicked
+      // nothing to do
+    }).catch(() => {
+      // if we get here, Cancel was clicked
+      // ignore changes
+      this.haveChanges = false
+      // route to dashboard
+      this.navigateToBusinessDashboard(this.getIdentifier)
+    })
+  }
+
+  @Watch('notation')
+  async onNotationChanged (): Promise<void> {
+    // if this is a court order and notation has changed, re-validate file upload component
+    if (this.isCourtOrder && this.enableValidation) {
+      await this.$nextTick() // wait for variables to update
+      this.$refs.fileUploadRef?.validate()
+    }
+  }
+
+  // NB: watch "fileKey" because it changes according to file validity (while "file" might not)
+  @Watch('fileKey')
+  async onFileKeyChanged (): Promise<void> {
+    // if this is a court order and file has changed, re-validate notation form
+    if (this.isCourtOrder && this.enableValidation) {
+      await this.$nextTick() // wait for variables to update
+      this.$refs.notationFormRef?.validate()
+    }
+  }
+
+  openFileDialog () {
+    // Access the underlying file input element and trigger click
+    const fileInput = this.$refs.fileUploadRef.$el.querySelector('input')
```

Comment:
Oh, so you're not using the new file uploader component?

How about moving this code to a public function in the file upload component, that you just call here?

Reason: this component should not know what's inside the file upload component.

Reply:
You didn't fix this here so it will have to be updated later.